hgtv/views/playlist.py

```diff
@@ -33,76 +35,64 @@ def process_playlist(playlist, playlist_url):
-                                        stream_playlist.videos.append(new_video)
-                            else:
-                                video = Video(playlist=playlist if playlist is not None else stream_playlist)
-                                video.title = item['title']['$t']
-                                video.video_url = item['media$group']['media$player']['url']
-                                if 'media$description' in item['media$group']:
-                                    video.description = markdown(item['media$group']['media$description']['$t'])
-                                for video_content in item['media$group']['media$thumbnail']:
-                                    if video_content['yt$name'] == 'mqdefault':
-                                        thumbnail_url_request = requests.get(video_content['url'])
-                                        filestorage = return_werkzeug_filestorage(thumbnail_url_request,
-                                            filename=secure_filename(item['title']['$t']) or 'name-missing')
-                                        video.thumbnail_path = thumbnails.save(filestorage)
-                                video.video_sourceid = item['media$group']['yt$videoid']['$t']
-                                video.video_source = u"youtube"
-                                video.make_name()
-                                playlist.videos.append(video)
+                api_key = app.config['YOUTUBE_API_KEY']
+                youtube = build('youtube', 'v3', developerKey=api_key)
+                playlistitems_list_request = youtube.playlistItems().list(
+                    playlistId=playlist_id,
+                    part='snippet',
+                    maxResults=50
+                )
+                playlist_info_request = youtube.playlists().list(
+                    id=playlist_id,
+                    part='snippet'
+                )
+                if playlist_info_request:
+                    playlist_infos = playlist_info_request.execute()
+                    for playlist_info in playlist_infos['items']:
+                        playlist.title = playlist.title or playlist_info['snippet']['title']
+                        if playlist_info['snippet']['description']:
+                            playlist.description = playlist_info['snippet']['description']
+                while playlistitems_list_request:
+                    playlistitems_list_response = playlistitems_list_request.execute()
+                    for playlist_item in playlistitems_list_response['items']:
+                        with db.session.no_autoflush:
+                            videos = Video.query.filter_by(video_source=u'youtube', video_sourceid=playlist_item['snippet']['resourceId']['videoId']).all()
+                        if videos:
+                            # If video isn't present in current playlist, copy the video parameters
```

Comment:
Change comment: `# If video is already in db but not in the current playlist, add it`